cornell-dti/course-plan
# Add Material Science Engineering and PAM majors

### Summary 

This pull request adds the Material Science Engineering and PAM majors to CoursePlan

- [x] Added material science major
- [x] Added PAM major
- [x] Fixed a credit bug in the human ecology college (3 -> 9 credits)
- [x] Added TODO to ling major I missed, fixed a description in human development major

![image](https://user-images.githubusercontent.com/25535093/115120331-ef076d80-9f7a-11eb-89fd-27a4de833440.png)

![image](https://user-images.githubusercontent.com/25535093/115122528-5840ae00-9f86-11eb-9139-6c1bd24fb147.png)

### Remaining TODOs

Put TODOs in the files where there were requirements we could not handle. Please confirm that these are all currently things that we cannot check.

### Test Plan 

1. Try to fulfill the entire Material Science Engineering Major and confirm the requirements match (especially the toggleable req and category req): https://www.mse.cornell.edu/mse/programs/undergraduate-programs/major/major-curriculum
2. Do the same for PAM here: https://www.human.cornell.edu/sites/default/files/Academics/Registrar/Curriculum%20sheets/2020-2021/PAM%20Curriculum%20Sheet%202020-2021.pdf
3. Confirm the credit bug in human ecology for the credits outside of major requirement is fixed.

### Notes 

Just had a few things to specifically mention
1. For the 2 categories requirement for MSE, I based the code off of the liberal studies code for engineering. However, I struggled with the ... operator and understanding what exactly the code was doing, so let me know if you have any suggestions on how to write that checker better!
2. The chart for the 'Natural Science 1' requirement in PAM is a little confusing. Do you think this means any course in this entire chart can count (like I coded)?

![image](https://user-images.githubusercontent.com/25535093/115122131-5249cd80-9f84-11eb-871e-abf9edb590bd.png)

3. Been noticing a lot of the human development requirements are similar across human ecology majors (Natural Science I in human development is a subset of the one in PAM, Natural Science II, Additional Requirements, Electives is exactly the same as human development, etc.). Might be worth factoring these out in some way or adding common checkers?
